### ISSUE TITLE
Is it possible to remove the vulnerability introduced by package node-fetch?

### ISSUE THREAD

**Issue by evansrobert** [vercel/next.js#27886]
### What version of Next.js are you using?

next@9.5.5

### What version of Node.js are you using?

11.2.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

 next export

### Describe the Bug

security reports about dependencies

Hi, @ijjk @timneutkens, I stumbled upon a vulnerability introduced by package **node-fetch@2.6.0**:
### Issue Description
When I build my project, I notice that a vulnerability [**CVE-2020-15168**](https://snyk.io/vuln/SNYK-JS-NODEFETCH-674311) detected in package **node-fetch(<2.6.1,>=3.0.0-beta.1 <3.0.0-beta.9)** is transitively referenced by **_next@9.5.5_**.
   However,  **next@9.5.5** is so popular that a large number of latest versions of active and popular downstream projects depend on it ([75,142 downloads per week](https://www.npmjs.com/package/next/v/9.5.5?activeTab=versions) and about **658 downstream projects**, e.g., @sentrei/common 1.131.0, @sentrei/web 1.131.0, @sentrei/ui 1.131.0, @sentrei/functions 1.131.0, @sentrei/video 1.131.0, etc.). 
In this case, the vulnerability [**CVE-2020-15168**](https://snyk.io/vuln/SNYK-JS-NODEFETCH-674311) can be propagated into these downstream projects and expose security threats to them.
   As you can see, **_next@9.5.5_** is introduced into the above projects via the following package dependency paths:
(1)`@sentrei/ui@1.131.0 ➔ @sentrei/common@27.3.0 ➔ next@9.5.5 ➔ @ampproject/toolbox-optimizer@2.6.0 ➔ cross-fetch@3.0.5 ➔ node-fetch@2.6.0`
    **......**

I know that it's kind of you to have removed the vulnerability since **_next@10.0.4_**. But, in fact, the above large amount of downstream projects cannot easily upgrade **next** from version **_9.5.5_** to _**(>=10.0.4)**_:
    The projects such as **@sentrei/common**, which introduced **next@9.5.5**, are not maintained anymore. These unmaintained packages can neither upgrade **next** nor be easily migrated by the large amount of affected downstream projects.  

   **Given the large number of downstream users, is it possible to release a new patched version with the updated dependency to remove the vulnerability from package next@9.5.5?**

### Suggested Solution

Since these inactive projects set a version constaint **9.5.\*** for **next** on the above vulnerable dependency paths, if **next** removes the vulnerability from **9.5.5** and releases a new patched version **_next@9.5.6_**, such a vulnerability patch can be automatically propagated into the downstream projects.

In **_next@9.5.6_**, maybe you can try to perform the following upgrade(not crossing major version):
`@ampproject/toolbox-optimizer 2.6.0 ➔ 2.7.1`;  
**Note:**
**_@ampproject/toolbox-optimizer@2.7.1(>=2.7.1)_** transitively depends on **node-fetch@2.6.1** which has fixed the vulnerability ([**CVE-2020-15168**](https://snyk.io/vuln/SNYK-JS-NODEFETCH-674311)).
Of course, you are welcome to share other ways to resolve the issue.^_^

**Comment by stefanprobst** [comment]
can you use [`resolutions`](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) to force v2.6.1?

**Comment by evansrobert** [comment]
@stefanprobst Such a fix is the additional efforts that npm community brings to you.
The vulnerability patch innext@9.5.6 can be automatically propagated into 658 projects  ([75,142 downloads per week](https://www.npmjs.com/package/next/v/9.5.5?activeTab=versions).
Literally, it indeed benefits a huge amount of downstream users.
Thanks again.

**Comment by ijjk** [comment]
Hi, this should be resolved in the latest version of Next.js, note `node-fetch` is not used client-side which is the only version of fetch being used with `next export`.

**Comment by godmar** [comment]
I came to this bug report because npm is reporting this vulnerability for me in next-12.0.8.

I believe that this issue may not be fixed except in canary as of today. The [advisory](https://github.com/advisories/GHSA-r683-j2x4-v87g) states that node-fetch prior to 2.6.7 is vulnerable, and next wasn't updated to require node-fetch@2.6.7 until [4 days ago](https://github.com/vercel/next.js/pull/33466/commits/fbfb6ae85907fdce146539cf01f41db6b5de2319). The package.json file contained [in release 12.0.8 uses 2.6.1](https://github.com/vercel/next.js/blob/v12.0.8/package.json#L126).

I believe that no action is necessary at this point - we can wait for the canary to be released in 12.0.9, but I'm leaving this note here in case anyone else worries about the npm audit warning. Mine looks like this:
```
node-fetch  <2.6.7
Severity: high
node-fetch is vulnerable to Exposure of Sensitive Information to an Unauthorized Actor - https://github.com/advisories/GHSA-r683-j2x4-v87g
fix available via `npm audit fix --force`
Will install next@10.0.1, which is a breaking change
node_modules/node-fetch
  next  9.0.6-canary.0 - 9.3.4-canary.0 || 10.0.2-canary.0 - 12.0.8
  Depends on vulnerable versions of node-fetch
  node_modules/next
    eslint-config-next  >=10.2.1-canary.2
    Depends on vulnerable versions of next
    node_modules/eslint-config-next
```
Also, do not run `npm audit fix --force` - this will downgrade your next.js to 10.0.1.

**Comment by quyctd** [comment]
@ijjk I'm using next.js 10 in production env, and upgrading Next.js to the latest version is quite risky because there is a lot of breaking changes. Do you know any options that don't requỉed an upgrade?
